### ISSUE TITLE
Build breaks upon renaming or deleting dependencies

### ISSUE THREAD

**Issue by davidwarrington** [damorej-at-theinstitutes/webpack-concat-files-plugin#12]
I believe the issue is with this line: https://github.com/damorej-at-theinstitutes/webpack-concat-files-plugin/blob/master/index.js#L66

I'm using this plugin in a project along with Sass, and if I rename or delete an scss file, even though the modules are unrelated to what I'm using this plugin for, I get an error.

```
/Users/dave/Projects/project/src/styles/05-components/supports.scss
(node:93074) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, lstat '/Users/dave/Projects/project/src/styles/05-components/supports.scss'
    at Object.lstatSync (fs.js:906:3)
    at /Users/dave/Projects/project/node_modules/webpack-concat-files-plugin/index.js:67:30
    at Array.filter (<anonymous>)
    at /Users/dave/Projects/project/node_modules/webpack-concat-files-plugin/index.js:65:12
    at _next3 (eval at create (/Users/dave/Projects/project/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:8:1)
    at eval (eval at create (/Users/dave/Projects/project/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:24:1)
```

Unfortunately I don't have time to dig much further into this right now, but I think the issue is caused by `fs.lstatSync(watchFile)` on said line returning null. In which case I believe the line needs to changed to something along the lines of:

```js
const fileStat = fs.lstatSync(watchFile);

if (!fileState) {
	return;
}

return !(fileStat.isDirectory());


// ...or, this might work in the latest versions of Node

return !(fs.lstatSync(watchFile)?.isDirectory());
```

**Comment by damorej-at-theinstitutes** [comment]
Thanks David! Your explanation makes sense. I'll take a closer look and try to have a fix published soon.

**Comment by davidwarrington** [comment]
Cheers @damorej-at-theinstitutes. I was thinking some more last night, and it might actually be better to do a more explicit `fs.existsSync` check before `lstatSync` just so it's more obvious what's happening.

Perhaps something like:
```js
return fs.existsSync(watchFile) && fs.lstatSync(watchFile).isDirectory()
```

It's probably a bit less efficient, but I think the intention is clearer?

**Comment by damorej-at-theinstitutes** [comment]
Hi @davidwarrington, sorry for the hold up on this. I just pushed a commit to the `develop` branch that should fix the issue (I basically implemented the suggestion in your most recent comment).

Would you be willing to confirm whether this fixes your issue? You can do so by temporarily updating your `package.json`'s dependency entry followed by `npm install` or similar:

`"webpack-concat-files-plugin": "damorej-at-theinstitutes/webpack-concat-files-plugin#develop"`

If you can confirm that the change resolves your issue, I'll publish a new release with that fix included.

Thanks again!

**Comment by damorej-at-theinstitutes** [comment]
As a side note, I've been considering adding a "Contributors" section to this repo's README.md to call out and thank folks who've contributed. Would you mind being included there if I were to add that section?

**Comment by davidwarrington** [comment]
@damorej-at-theinstitutes Sorry! I forgot to get back to you on this. After leaving my last comment I modified the file in my `node_modules` folder to use this and it's been fine ever since, so I'm confident that this works.

```js
return fs.existsSync(watchFile) && !fs.lstatSync(watchFile).isDirectory()
```

Cheers for actually making the commit though. And yes, I'm more than happy to be included within the contributors section of the readme. Thanks for asking 👍

**Comment by damorej-at-theinstitutes** [comment]
Thanks again @davidwarrington! I just published these changes to [npm as v0.4.1](https://www.npmjs.com/package/webpack-concat-files-plugin).

I'm going to close this issue now, but if we notice any issues we can reopen it or open a new one.
